packages/block-library/src/post-hierarchical-terms/index.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * Server-side rendering of the `core/post-hierarchical-terms` block.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Renders the `core/post-hierarchical-terms` block on the server.
+ *
+ * @param array    $attributes Block attributes.
+ * @param string   $content    Block default content.
+ * @param WP_Block $block      Block instance.
+ * @return string Returns the filtered post hierarchical terms for the current post wrapped inside "a" tags.
+ */
+function render_block_core_post_hierarchical_terms( $attributes, $content, $block ) {
+	if ( ! isset( $block->context['postId'] ) || ! isset( $attributes['term'] ) ) {
+		return '';
+	}
+
+	$post_hierarchical_terms = get_the_terms( $block->context['postId'], $attributes['term'] );
+	if ( empty( $post_hierarchical_terms ) ) {
+		return '';
+	}
+
+	$align_class_name = empty( $attributes['textAlign'] ) ? '' : ' ' . "has-text-align-{$attributes['textAlign']}";
+
+	$terms_links = '';
+	foreach ( $post_hierarchical_terms as $term ) {
+		$terms_links .= sprintf(
+			'<a href="%1$s">%2$s</a> | ',
+			get_term_link( $term->term_id ),
+			esc_html( $term->name )
+		);
+	}
+	$terms_links = trim( $terms_links, ' | ' );
+
+	return sprintf(
+		'<div class="%1$s">%2$s</div>',
+		'wp-block-post-hierarchical-terms' . esc_attr( $align_class_name ),
```

Comment:
small note that`wp-block-hierarchical-terms` class should be automatically added once #24546 lands.